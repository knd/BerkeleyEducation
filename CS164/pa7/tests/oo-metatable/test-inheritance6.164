def Insect = {}
Insect.__index = Insect
Insect.numLegs = lambda(self) { 6 }

def Spider = {}
setmetatable(Spider, Insect)
Spider.numLegs = lambda(self) { 8 }
Spider.up = lambda(self) {
  getmetatable(self).__index
}

print "Spider has: "
print Spider:numLegs()
print "Insect has: "
print Spider:up().numLegs(Spider)
