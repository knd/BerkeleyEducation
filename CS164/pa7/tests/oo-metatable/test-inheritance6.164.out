Spider has: 
8
Insect has: 
6
